src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +88,59 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
+
+            var localizedType = context.Key.ContainerType ?? context.Key.ModelType;
+            var localizer = _stringLocalizerFactory?.Create(localizedType);
+
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Description) &&
+                    displayAttribute.ResourceType == null)
+                {
+                    displayMetadata.Description = () => localizer[displayAttribute.GetDescription()].Value;
+                }
+                else
+                {
+                    displayMetadata.Description = () => displayAttribute.GetDescription();
+                }
             }
 
-            // DisplayFormatString
-            if (displayFormatAttribute != null)
+            // Placeholder
+            if (displayAttribute != null)
             {
-                displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Prompt) &&
+                    displayAttribute.ResourceType == null)
+                {
+                    displayMetadata.Placeholder = () => localizer[displayAttribute.GetPrompt()];
```

Comment:
Why no `.Value` here and on [line 132](https://github.com/aspnet/Mvc/pull/5129/files#diff-cf1eab5c65bcfd6b66eecf4662e74ea4R132). If it's incorrect, remove it on [line 102](https://github.com/aspnet/Mvc/pull/5129/files#diff-cf1eab5c65bcfd6b66eecf4662e74ea4R102).